nixos/modules/services/games/minecraft-server.nix

```diff
@@ -210,6 +237,25 @@ in {
           rm .declarative
         fi
       '');
+
+      preStop = let 
+        script = pkgs.writeScript "minecraft-stop-commands" cfg.stopCommands;
```

Comment:
You can use `pkgs.writeShellScript` here instead, this also makes sure there are no syntax errors in the script